DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/streaming_data_interfaces/src/streaming_data_interfaces.cpp

```diff
@@ -0,0 +1,119 @@
+#include <iostream>
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/ext/intel/prototype/pipes_ext.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+#define THRESHOLD 200
+
+// Forward declare the kernel and pipe names
+// (this prevents unwanted name mangling in the optimization report)
+class InPixel;
+class OutPixel;
+class Threshold;
+
+// StreamingBeat struct to support Avalon packet sideband signals
+using StreamingBeatT = sycl::ext::intel::experimental::StreamingBeat<
+    unsigned short, true, false>;
```

Comment:
```suggestion
using StreamingBeatT = sycl::ext::intel::experimental::StreamingBeat<
    unsigned short, // type carried over this Avalon streaming interface's data signal
    true,         // enable startofpacket and endofpacket signals
    false>;      // disable the empty signal
```
elaborate magic numbers